Modules/xASL_module_ASL.m

```diff
@@ -874,3 +512,305 @@
+	x.modules.asl.DummyScanPositionInASL4D = [];
+end
+if ~isfield(x.modules.asl,'M0PositionInASL4D') 
+	x.modules.asl.M0PositionInASL4D = [];
+end
+if ~isfield(x.modules.asl,'motionCorrection')
+    x.modules.asl.motionCorrection = 1;
+end
+if ~isfield(x,'DoWADQCDC')
+    x.DoWADQCDC = false; % default skip WAD-QC stuff
+end
+
+
+%% 3. Multi-parameter parsing
+parNames = {'EchoTime' 'Initial_PLD' 'LabelingDuration'};
+parAbbreviation = {'TE' 'PLD' 'LD'};
+parTolerance = {0.001 0 0};
+parLowestValue = {0 0 0};
+
+for iPar=1:length(parNames)
+
+    if ~isfield(x.Q, parNames{iPar})
+        % if the field is completely missing
+        warning(['Missing field x.Q.' parNames{iPar} ', this may be needed for quantification']);
+        % After the warning, we default to single-parameter processing
+        fprintf('%s\n', ['Defaulting to single-' parAbbreviation{iPar} ' ASL processing']);
+	    x.Q.(['Number' parNames{iPar}]) = 1;
+        % PM: rename this parameter to x.Q.nParName
+        x.Q.(['bQuantifyMulti' parAbbreviation{iPar}]) = false;
+    
+    elseif isempty(x.Q.parNames{iPar}) || ~isnumeric(x.Q.parNames{iPar})
+        % if the field exists but is illegal
+        warning(['Illegal field x.Q.' parNames{iPar} ', this should not be empty and should contain numerical values']);
+        % After the warning, we default to single-parameter processing
+        fprintf('%s\n', ['Defaulting to single-' parAbbreviation{iPar} ' ASL processing']);
+	    x.Q.(['Number' parNames{iPar}]) = 1;
+        x.Q.(['bQuantifyMulti' parAbbreviation{iPar}]) = false;
+    else
+        % Check, with allowed tolerance (0 is without tolerance) what the unique parameters are
+        x.Q.(['unique' parNames{iPar}]) = uniquetol(x.Q.(parNames{iPar}), parTolerance);
+    
+        x.Q.(['Number' parNames{iPar}]) = length(x.Q.(['unique' parNames{iPar} 's']));
+        % Obtain the number of unique parameters
+    
+        if sum(x.Q.(parNames{iPar})<=parLowestValue{iPar})>0
+            warning(['x.Q.' parNames{iPar} ' should be larger than ' num2str(parLowestValue{iPar})]);
+        end
+    
+        % Handle different number of parameters (== potentially different ASL sequences)
+        if x.Q.(['Number' parNames{iPar}]) == 1
+            fprintf('%s\n', ['Single-' parAbbreviation{iPar} ' ASL detected']);
+        elseif x.Q.(['Number' parNames{iPar}]) == 2
+            fprintf('%s\n', ['Dual-' parAbbreviation{iPar} ' ASL detected']);
+            
+            if strcmp(parAbbreviation{iPar}, 'TE')
+                warning('Dual-echo ASL detected, dual-echo ASL processing not yet implemented');
+                fprintf('%s\n', 'We will process these ASL images as if they have a single echo, using the first echo only');
+                fprintf('%s\n', 'Assuming that this is a dual-echo ASL+BOLD sequence');
+    
+                x.Q.EchoTime = min(x.Q.EchoTime);
+		    % Update the other fields as well
+		     x.Q.uniqueEchoTimes = x.Q.EchoTime;
```

Comment:
uniqueEchoTime
NumberEchoTime

Please search and replace that in the entire ExploreASL code.

Reply:
Done in bf3e623cce3732ff0b889c878d9f2517d428da5f